lib/spack/spack/cmd/info.py

```diff
@@ -237,6 +239,62 @@ def print_text_info(pkg):
     else:
         color.cprint("    None")
 
+    # Some built-in base packages (e.g., Autotools) define callback (e.g.,
+    # check) inherited by descendant packages. These checks may not result
+    # in build-time testing if the package's build does not implement the
+    # expected functionality (e.g., a 'check' or 'test' targets).
+    #
+    # So the presence of a callback in Spack does not necessarily correspond
+    # to the actual presence of built-time tests for a package.
+    for callbacks, phase in [(pkg.build_time_test_callbacks, 'Build'),
+                             (pkg.install_time_test_callbacks, 'Install')]:
+        color.cprint('')
+        color.cprint(section_title('Available {0} Phase Test Methods:'
+                                   .format(phase)))
+        if callbacks:
+            colify(sorted(callbacks), indent=4)
+        else:
+            color.cprint('    None')
+
+    # PackageBase defines an empty install/smoke test but we want to know
+    # if it has been overridden and, therefore, assumed to be implemented.
+    color.cprint('')
+    color.cprint(section_title('Stand-Alone/Smoke Test Methods:'))
+    names = []
+    if has_test_method(pkg):
+        pkg_base = spack.package.PackageBase
+        test_pkgs = [str(cls.test) for cls in inspect.getmro(pkg) if
```

Comment:
@tldahlgren I think you need `inspect.getmro(pkg.__class__)` here.

Reply:
I completely forgot about that method being a potential source of issues here. Thanks becker33!

Reply:
@becker33 Thanks but that change didn't help either. 😞